msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignInTest.kt

```diff
@@ -24,16 +24,87 @@
 package com.microsoft.identity.client.e2e.tests.network.nativeauth
 
 import com.microsoft.identity.internal.testutils.nativeauth.NativeAuthCredentialHelper
+import com.microsoft.identity.internal.testutils.nativeauth.api.TemporaryEmailService
 import com.microsoft.identity.nativeauth.statemachine.errors.SignInError
+import com.microsoft.identity.nativeauth.statemachine.errors.SubmitCodeError
 import com.microsoft.identity.nativeauth.statemachine.results.SignInResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignUpResult
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.StandardTestDispatcher
 import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
 import org.junit.Assert
+import org.junit.Before
 import org.junit.Test
+import org.robolectric.RuntimeEnvironment.application
 
 class SignInTest : NativeAuthPublicClientApplicationAbstractTest() {
 
+    private val tempEmailApi = TemporaryEmailService()
+
+    // Remove default Coroutine test timeout of 10 seconds.
+    private val testDispatcher = StandardTestDispatcher()
+
+    @Before
+    override fun setup() {
+        super.setup()
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    private fun <T> retryOperation(
```

Comment:
Looks like this is used in a few places - can we move it to `NativeAuthPublicClientApplicationAbstractTest`?

Reply:
That's a good point. I will move it to NativeAuthPublicClientApplicationAbstractTest after validating it works in the pipeline.